_tests/tests_suite_test.go

```diff
@@ -27,31 +27,55 @@ func init() {
 }
 
 func getRandAppName() string {
-	return fmt.Sprintf("apps-test-%d", rand.Int())
+	name := fmt.Sprintf("test-%d", rand.Int())
+	if len(name) > 24 {
+		name = name[:24]
+	}
+	return name
 }
 
 func TestTests(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "Tests Suite")
+	RunSpecs(t, "Deis Workflow")
 }
 
 var (
-	testAdminUser     = fmt.Sprintf("test-admin-%d", GinkgoConfig.RandomSeed)
-	testAdminPassword = "asdf1234"
-	testAdminEmail    = fmt.Sprintf("test-admin-%d@deis.io", GinkgoConfig.RandomSeed)
+	testAdminUser     = "admin"
+	testAdminPassword = "admin"
+	testAdminEmail    = "admin@example.com"
 	testUser          = fmt.Sprintf("test-%d", GinkgoConfig.RandomSeed)
 	testPassword      = "asdf1234"
 	testEmail         = fmt.Sprintf("test-%d@deis.io", GinkgoConfig.RandomSeed)
+	keyName           = fmt.Sprintf("deiskey-%v", GinkgoConfig.RandomSeed)
 	url               = getController()
+	debug             = os.Getenv("DEBUG") != ""
+	homeHome          = os.Getenv("HOME")
 )
 
+var testRoot, testHome, keyPath string
+
 var _ = BeforeSuite(func() {
 	// use the "deis" executable in the search $PATH
-	_, err := exec.LookPath("deis")
+	output, err := exec.LookPath("deis")
+	Expect(err).NotTo(HaveOccurred(), output)
+
+	testHome, err = ioutil.TempDir("", "deis-workflow-home")
 	Expect(err).NotTo(HaveOccurred())
+	os.Setenv("HOME", testHome)
 
 	// register the test-admin user
-	register(url, testAdminUser, testAdminPassword, testAdminEmail)
+	output, err = execute("deis register %s --username=%s --password=%s --email=%s", url, testAdminUser, testAdminPassword, testAdminEmail)
+
+	if strings.Contains(output, "must be unique") {
+		// Already registered
+		login(url, testAdminUser, testAdminPassword)
+	} else {
+		Expect(err).NotTo(HaveOccurred(), output)
+		Expect(output).To(SatisfyAll(
+			ContainSubstring("Registered %s", testAdminUser),
+			ContainSubstring("Logged in as %s", testAdminUser)))
+	}
+
```

Comment:
Tt would be better if this and the below execution logic (in `BeforeEach`) used the `start`/`gexec` flow. That's what we should be migrating to.